jupyterhub/apihandlers/base.py

```diff
@@ -131,19 +132,25 @@ def write_error(self, status_code, **kwargs):
             json.dumps({'status': status_code, 'message': message or status_message})
         )
 
-    def server_model(self, spawner, include_state=False):
+    def server_model(self, spawner, include_state=False, include_user_and_id=False):
         """Get the JSON model for a Spawner"""
-        return {
+        if isinstance(spawner, orm.Spawner):
+            user = self._user_from_orm(spawner.user)
+            spawner = spawner_obj.Spawner(orm_spawner=spawner, user=user)
+        server = {
             'name': spawner.name,
-            'last_activity': isoformat(spawner.orm_spawner.last_activity),
-            'started': isoformat(spawner.orm_spawner.started),
+            'last_activity': isoformat(spawner.last_activity),
+            'started': isoformat(spawner.started),
             'pending': spawner.pending,
             'ready': spawner.ready,
             'state': spawner.get_state() if include_state else None,
             'url': url_path_join(spawner.user.url, spawner.name, '/'),
             'user_options': spawner.user_options,
             'progress_url': spawner._progress_url,
         }
+        if include_user_and_id:
+            server.update({'id': spawner.id, 'user': {'name': spawner.user.name}})
```

Comment:
With `DELETE /server/{id}` out of the picture now, what do we need the ID for? To delete the server I need the username and the servername. The ID doesn't seem like something I'd need as a consumer?

If we do remove it we can rename this from the long name of `include_user_and_id` to `include_user`.

Reply:
Yeah `id` can be removed.